### PR TITLE
Refactor backup and restore to verify backup/restore using local snapshot

### PR DESCRIPTION
**Pull request by shashank855**
**What this PR does / why we need it**:

- Verify backup and restore using local snapshot.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: experiments/functional/backup_and_restore/backup-restore.yml [hunk 1]
```diff
@@ -8,14 +8,21 @@
...
 
       - name: Creating Backup using GCP bucket
         shell: >
           velero backup create {{ velero_backup_name }} --include-namespaces={{ app_ns }} --snapshot-volumes --volume-snapshot-locations=gcp-bucket
         args:
           executable: /bin/bash
-        when: bucket_type == "GCP"
+        when: bucket_type == "GCP" and lookup('env','LOCAL_SNAPSHOT') == "false"
+
+      - name: Creating Backup using GCP bucket
```
Comment (mynktl) [inline experiments/functional/backup_and_restore/backup-restore.yml:20]:
```suggestion
      - name: Creating local Backup using GCP bucket
```

File: experiments/functional/backup_and_restore/backup-restore.yml [hunk 2]
```diff
@@ -8,14 +8,21 @@
           velero backup create {{ velero_backup_name }} --include-namespaces={{ app_ns }} --snapshot-volumes --volume-snapshot-locations=minio
         args:
           executable: /bin/bash
-        when: bucket_type == "minio"
+        when: bucket_type == "minio" and lookup('env','LOCAL_SNAPSHOT') == "false"
 
       - name: Creating Backup using GCP bucket
```
Comment (mynktl) [inline experiments/functional/backup_and_restore/backup-restore.yml:13]:
```suggestion
      - name: Creating remote Backup using GCP bucket
```

**Review by mynktl — APPROVED** [review]
Changes are good. Given few minor comments.

**Review by nsathyaseelan — CHANGES_REQUESTED** [review]
@shashank855  can you please resolve the conflicts

**Review by nsathyaseelan — APPROVED** [review]
lgtm

**Pull request merged**